### DISCUSSION TITLE
Can we have a (CV) window splitter with a time-like window length?

### DISCUSSION THREAD

**Discussion by JeromeLovy** [alan-turing-institute/sktime#3427] [Q&A] [answered]
Hi,

Can we have a (CV) window splitter with a _time-like_ window length?

So far, I'm under the impression that the answer is **no**. Indeed, (CV) window splitters apparently expect arguments to be either all _time-like_ or all _.iloc friendly_ (see [_inputs_are_supported()](https://github.com/alan-turing-institute/sktime/blob/78913aaa034171e0980be98966cbfb162c0472e9/sktime/forecasting/model_selection/_split.py#L169)). A _time-like_ window length would thus imply a _time-like_ forecasting horizon. But it seems that for a forecasting horizon to be _time-like_, it has to be _absolute_ (as per the [ForecastingHorizon](https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.base.ForecastingHorizon.html) API). And finally, (CV) window splitters seem to reject _absolute_ forecasting horizons (see [_split.py:_check_fh()](https://github.com/alan-turing-institute/sktime/blob/main/sktime/forecasting/model_selection/_split.py#L64)).

Did I miss something?

PS: My motivation for using a _time-like_ window length is that I am trying to train a model on a training window without a specified frequency (typically a training window with data gaps in it). So far, my tests show me that combining such a training window with an *int* window length is unsupported as it raises a `TypeError: unsupported operand type(s) for *: 'NoneType' and 'int'` (see [SingleWindowSplitter._split()](https://github.com/alan-turing-institute/sktime/blob/86506e17539dfee3073a4eb36323909bc5789f99/sktime/forecasting/model_selection/_split.py#L1267) or [CutoffSplitter._split()](https://github.com/alan-turing-institute/sktime/blob/86506e17539dfee3073a4eb36323909bc5789f99/sktime/forecasting/model_selection/_split.py#L733)).

**Comment by khrapovs** [comment, marked as answer]
I have been working on this for quite some time already. At least some flexibility in this direction is already implemented, but not well documented. The umbrella issue is https://github.com/alan-turing-institute/sktime/issues/1737. You can take a look at the series of merged PRs and find some mentions of timedelta, etc. If you are brave enough to start contributing to this work stream, you are always welcome!

**Reply by JeromeLovy** [reply]
Thanks for your insight, @khrapovs! (The discussion under the #1737 issue is very interesting btw.)

Should I take your answer for a confirmation that splitters currently (as of v0.13.1) don't accept a *time-like* window length?

**Reply by khrapovs** [reply]
They do! But since I did not document it yet, you should, unfortunately, check yourself by reading through the test cases here: https://github.com/alan-turing-institute/sktime/blob/main/sktime/forecasting/model_selection/tests/test_split.py. In particular, take a note on the variables like `TEST_WINDOW_LENGTHS`. Some of them have time-like values.

**Reply by fkiraly** [reply]
@khrapovs, @JeromeLovy, maybe someone (looking at no one in particular here 😉 ) could be so kind and document it?

**Reply by JeromeLovy** [reply]
Thanks for the pointer to the tests, @khrapovs!

I might try to run these *test_split* test cases step by step to better understand how things are working. (I would have to manage to launch the tests with **pytest**, which I wasn't able to so far 😅. Not sure if I'll have the time to continue investigating on this.)

**Reply by khrapovs** [reply]
Actually, wait a sec. It is even documented! See for example https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.model_selection.SlidingWindowSplitter.html. `window_length: int or **timedelta** or **pd.DateOffset**`. What I meant is that there is no notebook in examples specific to this functionality, but the API reference is quite up to date.

**Reply by fkiraly** [reply]
well, thank you, anonymous documenter :-)
